storage/redistore/peer_crud.go

```diff
@@ -0,0 +1,70 @@
+package redistore
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/chihaya/chihaya/bittorrent"
+	"github.com/garyburd/redigo/redis"
+)
+
+func decodePeerKey(pk string) bittorrent.Peer {
+	return bittorrent.Peer{
+		ID:   bittorrent.PeerIDFromString(string(pk[:20])),
+		Port: binary.BigEndian.Uint16([]byte(pk[20:22])),
+		IP:   net.IP(pk[22:]),
+	}
+}
+
+func addAndCleanPeer(s *peerStore, infoHash bittorrent.InfoHash, peerType string, pk serializedPeer) error {
+	nameSpacedKey := addNameSpace(fmt.Sprintf("%s%s", peerType+":", infoHash))
+	s.conn.Send("MULTI")
+	s.conn.Send("ZADD", nameSpacedKey, time.Now().Unix(), pk)
+	s.conn.Send("EXPIRE", nameSpacedKey, s.gcValidity)
```

Comment:
I did not do that because I wanted to keep the two functionalities independent of each other. For example, I can have a peerlife time for every peer in zadd that will be removed during removePeers. But what if the peer client gets closed before it issues a removePeers? In this case the gc will be the expire for all holders. Does this seem reasonable? If not I can put them both together.